modules/programs/hyprpaper.nix

```diff
@@ -0,0 +1,108 @@
+{ pkgs, config, lib, ... }:
+
+with lib;
+
+let cfg = config.programs.hyprpaper;
+in {
+  meta.maintainers = [ lib.maintainers.silmarp ];
+
+  options = {
+    programs.hyprpaper = {
+      enable = mkEnableOption "hyprpaper";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.hyprpaper;
+        defaultText = literalExpression "pkgs.hyprpaper";
+        description = "The package to use for the hyprpaper binary.";
+      };
+
+      systemd = {
+        enable =
+          mkEnableOption "Whether to enable hyprpaper systemd integration.";
+
+        target = mkOption {
```

Comment:
This seems unnecessary. Any session that could run this service should already bind to `graphical-session.target`.